### ISSUE TITLE
Humble report after using b5 Live ISO and suggestion

### ISSUE THREAD

**Issue by lfom** [instantOS/instantOS#73]
**Describe the bug**
After booting live media, screen is rotated 90 degrees anti-clockwise, and it is not possible to use any terminal emulator.

**To Reproduce**
1. Download ISO from GitHub
2. Copy ISO to Ventoy drive
3. Boot InstantOS Live ISO
4. Screen is rotated and no terminals work

**Expected behavior**
I could rotate the screen using a tools provided in the live media, but this could be avoided (more info below). It should be possible to run commands in the provided terminal when using the Live media for testing purposes.

**Screenshots**
None.

**Device (only relevant in some cases):**
 - GPU: Intel HD
 - CPU: Atom x5 Z8350
 - Manufacturer/Model if prebuilt of laptop: Lenovo Miix-320

**Additional context**
The device is a convertible and the display is reported as 800x1280 (it can also be used as tablet), it is fixed in some distros I have used but it seems that the kernel used in Arch does not handle it fine (what is weird IMO). But I could rotate the display to the proper orientation using a tool provided in Settings.
But none of the terminals works: they would open a window, I could see what I typed but hitting ENTER did nothing. I could login to a TTY (that was displayed correctly, strangely) to check the info I wanted, like the kernel and XOrg drivers.
It is a real pity that InstantOS/WM does not support Wayland: the Intel driver is really bad, with lots of tearing. It can be somewhat improved with TearFree mode + doublebuffer, but stil...
And the tools provided are nice, only the keyboard shortcut map is... far below everything else InstantOS provides. I know InstantOS isn't meant for noobs, but... Just a suggestion.
Anyway, congrats for the great WM and efforts so far.
Regards.

**Comment by con-f-use** [comment]
The terminal emulator part was likely a bug in the live iso, that is already fixed. The rotation thing is a job for @paperbenni

**Comment by con-f-use** [comment]
As to your suggestions. Wayland does not yet support a few key features (Nvidia, ...) and does things differently enough, that a substantial re-write of instantWM would be required. So there will be no Wayland version for the foreseeable future, because of the development speed of Wayland. The keyboard shortcut system will be reworked, at latest for a 1.0 release, allowing to customize them without re-compiling instantWM. But I would call the defaults pretty good actually, keep in mind that they need to work for a number of keyboard layouts, languages and not collide with shortcuts of popular software. Given these constrains, I wouldn't call them "far below" anything.

**Comment by lfom** [comment]
Hello, there. Thank you for your reply. I think I have worded two things incorrectly or not clear enough:
1. It seems that the kernel handles the display rotation fine as the TTY was in the right orientation while the display was attached to the keyboard, so the problem is in XOrg.
2. The problem I see related to the keyboard shortcuts isn't the chosen key combos themselves, but how they are displayed/shown.
Thanks again. Reagrds.

**Comment by con-f-use** [comment]
Still not sure what you mean in 2., are you dissatisfied with the discoverability of the shortcut map at <kbd>Super</kbd> + <kbd>F1</kbd> and in the top left corner menu, or with its correctness or with its grouping?

**Comment by lfom** [comment]
Nope, only the presentation itself: IMHO a simple `less` in a terminal window is much simpler than the other utilities provided in InstantOS. Just for better visualization, what is used in regolith looks much better and is more useful.

**Comment by ppenguin** [comment]
> Nope, only the presentation itself: IMHO a simple `less` in a terminal window is not at the level of the other utilities provided in InstantOS. Just for better visualization, what is used in _regolith_ looks much better and is more useful IMO.

I'm not familiar with *regolith*, but coming to *instantOS* from *awesomewm* it immediately struck me too. The *"shortcut cheatsheet"* in *awesome* is beautiful and offers a very clear overview compact enough for one screen, so it would be a good example IMHO.

**Comment by con-f-use** [comment]
Can you post pictures or Videos for reference?

**Comment by lfom** [comment]
1. Screen rotation: I have tested many distros using the same machine, and it seems that kernels from 5.4 at least work fine for the console/TTY. The screen rotation is a problem with Intel drivers or the open source _modesetting_ one. The only way it works out of the box is with GDM3 + Wayland (and a newer kernel also, probably it needs the accelerometer to detect the screen rotation, I am not sure). Other than that, on every distro I tried I had to use `xrandr` to rotate the display manager and DE.

2. Shortcuts window: For reference about what I said, a screen shot from regolith (from its home page).
![regolith-screenshot-intro](https://user-images.githubusercontent.com/15659999/103177043-7eb2d000-4855-11eb-9d2a-929e23d2cd0f.png)

**Comment by ppenguin** [comment]
FWIW: For me rotation on Manjaro only worked with KDE/Plasma on a convertible (HP spectre x360) without any configuration, not with Gnome or instantOS.

I finally resorted to making [a fork of 2in1screen](https://github.com/ppenguin/2in1screen) inspired by its mention https://wiki.archlinux.org/index.php/Tablet_PC (which mentions also that Gnome should work but it didn't).

Of course it also uses `xrandr`, which in my experience is the way to go as a backend for all display config related functionality if using a non-desktop-environment. (I've tried many alternatives like "hybrid xsessions" and so on to be able to piggyback off DE-config functionality but it's really too cumbersome and unstable to get it working, not to mention the bloated dependencies).

Interestingly, `2in1screen` is also "suckless" like, so I'd recommend it for integration in instantOS too.

**Comment by ppenguin** [comment]
> Can you post pictures or Videos for reference?
awesome cheat sheet: (sorry, it's a DT video, but I don't use awesome anymore so that's what I could find easily :stuck_out_tongue_winking_eye: )
https://youtu.be/qKtit_B7Keo?t=904

**Comment by con-f-use** [comment]
Meh, I'm underwhelmed.

**Comment by ppenguin** [comment]
I agree that the searching is a nice feature, but the fact that an (arbitrarily sorted) window content is just arbitrarily clipped is also very unusual from a UX perspective, especially since the window is almost screen filling but has only very limited width for the payload. It may be more intuitive if one could see a scroll bar, otherwise it kind of "looks broken", just "UI-intuitively" speaking. Another solution would be to allow some kind of re-layout to have the table side by side, and augmented with coloured grouping somehow, while retaining search functionality. But that would be a significantly different implementation I'm afraid.

One issue with searching is that it's very sensitive to using correct keywords. For example: there is actually a lot of functionality related to switching floating/non-floating, but only a few hits when searching "float". Of course changing the content of the descriptions would go a long way there too, but categorising and a full overview seem more natural there.

In the end it's a similar discussion as about "what colour should something have", so not really worth a lot of effort or at least I'd say a lower priority...

**Comment by paperbenni** [comment]
A scrollbar for instantmenu is on the todo for beta7 but will probably take some time to integrate.
I agree that subcategories for the shortcuts would be useful, colors should also be doable

**Comment by paperbenni** [comment]
as for the 2in1screen I'd be down with integrating it into instantOS. I don't have the hardware to test with so I'd need your help for that but it's a welcome addition given you're ready to maintain the tool for the time being

**Comment by ppenguin** [comment]
Sure, my time is quite limited from now (I'll have to focus on my startup), but in a limited fashion I can contribute to that. (I'm not sure yet though how to make it work in a generic fashion, since the specification of the pointer devices which must have their coordinate-system transformed is quite specific and for now hard-coded. One could select per ID, but still need to find out how to select/filter the needed IDs).
